internal/exec/stages/disks/luks.go

```diff
@@ -391,6 +392,15 @@ func (s *stage) reuseLuksDevice(luks types.Luks, keyFilePath string) error {
 		}
 	}
 
+	dump, err := exec.Command(distro.CryptsetupCmd(), "luksDump", devAlias).CombinedOutput()
+	if err != nil {
+		return err
+	}
+	allowsDiscards := regexp.MustCompile("\nFlags:.+allow-discards").Match(dump)
+	if allowsDiscards != util.IsTrue(luks.Discard) {
+		return fmt.Errorf("volume allow-discards flag %v doesn't match expected value %v", allowsDiscards, util.IsTrue(luks.Discard))
+	}
```

Comment:
There's `cryptsetup luksDump --dump-json-metadata` which includes the flags. Should we use that instead? (Assuming the switch isn't super recent.)

Reply:
RHEL 8 doesn't have it, but luckily we don't need to care about that anymore.